### PR TITLE
71 initial iota streams integration

### PR DESCRIPTION
**Pull request by dominic22**
- subscription management in streams
- initial read/write from to channel

Info:
- will add tests later since we need to have the features in end of may

### CODE REVIEW

File: api/src/services/streams-service.ts [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+		seed?: string
+	): Promise<{ seed: string; subscriptionLink: string; subscriber: Subscriber }> => {
+		const annAddress = streams.Address.from_string(announcementLink);
+		const options = new streams.SendOptions(1, true, 1);
+
+		if (!seed) {
+			seed = this.makeSeed(81);
+		}
+
+		const subscriber = new streams.Subscriber(this.node, seed, options);
```
[the remark this answers is not in the document]
Comment (sobolev-alexey) [reply api/src/services/streams-service.ts]: so, you would like to keep the current Subscriber where possible by providing the seed, and generate a new seed as a backup solution?

File: api/src/services/channel-service.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+		this.password = config.statePassword;
+	}
+
+	create = async (userId: string, topics: Topic[], seed?: string): Promise<{ seed: string; channelAddress: string }> => {
+		const res = await this.streamsService.create(seed);
+
+		const subscription: Subscription = {
+			userId,
+			type: SubscriptionType.Author,
+			channelAddress: res.channelAddress,
```
[the remark this answers is not in the document]
Comment (sobolev-alexey) [reply api/src/services/channel-service.ts:34]: Where this addressed somewhere, do do you have a validation on an earlier stage?

[the remark this answers is not in the document]
Comment (dominic22) [reply api/src/services/channel-service.ts:34]: added a check here 👍

File: api/src/services/streams-service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dominic22) [reply api/src/services/streams-service.ts]: I want to give the client the option to decide whether to create a new subscriber for this channel or reuse a subscriber by providing the seed that's why it is optional to add in the request body

[the remark this answers is not in the document]
Comment (sobolev-alexey) [reply api/src/services/streams-service.ts]: ok, sounds good

**Review by sobolev-alexey — APPROVED** [review]

**Pull request merged**